zappa/handler.py

```diff
@@ -60,6 +60,20 @@ def __init__(self, settings_name="zappa_settings", session=None):
 
         # We haven't cached our settings yet, load the settings and app.
         if not self.settings:
+            zappa_settings_file = settings_name + ".py"
+            # Try loading from ZAPPA_SETTINGS environment variable if the file
+            # is not present. This will be the case for docker based deployments,
+            # see https://github.com/Miserlou/Zappa/pull/2192 for more details
+            if not os.path.exists(zappa_settings_file):
+                if os.environ.get('ZAPPA_SETTINGS'):
+                    with open(zappa_settings_file, 'w') as f_out:
+                        f_out.write(os.environ.get('ZAPPA_SETTINGS').replace('\\n', '\n'))
+                else:
+                    raise Exception(
+                        'Either a {} file or ZAPPA_SETTINGS '
+                        'environment variable must be present'.format(zappa_settings_file)
+                    )
+
```

Comment:
The other option to handling these settings involves creating the `zappa_settings.py` file as part of the Docker image creation. This is less ideal in my view and it is one more thing users have to remember to do when creating their Docker image, and it creates a 1-1 mapping between docker image and lambda function, when in reality you could use the same docker image for multiple lambda functions.